cmd/agent/app/reporter/grpc/flags.go

```diff
@@ -44,7 +44,10 @@ func (b *ConnBuilder) InitFromViper(v *viper.Viper) (*ConnBuilder, error) {
 	if err != nil {
 		return b, fmt.Errorf("failed to process TLS options: %w", err)
 	}
-	b.TLS = tls
+	if tls.Enabled {
+		tlsConf := tls.ToOtelClientConfig()
+		b.TLS = &tlsConf
+	}
```

Comment:
do we want to add a test here for the missing code coverage?

Reply:
done